presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergAbstractMetadata.java

```diff
@@ -343,9 +343,16 @@ public void addColumn(ConnectorSession session, ConnectorTableHandle tableHandle
         if (!column.isNullable()) {
             throw new PrestoException(NOT_SUPPORTED, "This connector does not support add column with non null");
         }
+
+        org.apache.iceberg.types.Type columnType = toIcebergType(column.getType());
+
+        if ((columnType.typeId() == Type.TypeID.TIMESTAMP) && columnType.toString().equals("timestamptz")) {
```

Comment:
Would something like this work?

```suggestion
        if ((columnType instanceOf TimestampType) && ((columnType (TimestampType)).shouldAdjustToUTC()) {
```

Reply:
`instanceOf `on iceberg type will not work. iceberg api code is using `shouldAdjustToUTC` in some cases. I have used `equals` after converting to `toIcebergType` on iceberg type which is readable.